Math/NumberTheory/Divisors.hs

```diff
@@ -0,0 +1,46 @@
+import System.IO
+import Data.Ratio
+
+
+mylog :: Integer -> Integer -> Integer
+mylog a b 
+	| ((a `mod` b) /= 0)  = 0
+	| otherwise = (1 + mylog (a `div` b) b)
+
+
+getfactors :: Integer -> Integer -> [(Integer,Integer)]
```

Comment:
I suggest using `Math.NumberTheory.Primes.factorise` to obtain a prime factorisation.